### PR TITLE
ARROW-14437: [Python] Make CSV cancellation test more robust

### PR DESCRIPTION
**Pull request by pitrou**
Size the workload according to the machine speed, to increase the probability of actually triggering cancellation.

### CODE REVIEW

**Comment by pitrou** [comment]
cc @kszucs @cyb70289

**Review by cyb70289 — APPROVED** [review]
LTGM

File: python/pyarrow/tests/test_csv.py [hunk 1]
```diff
@@ -1321,8 +1321,17 @@ def test_cancellation(self):
...
 
         # Make the interruptible workload large enough to not finish
-        # before the interrupt comes, even in release mode on fast machines
-        large_csv = b"a,b,c\n" + b"1,2,3\n" * 200_000_000
+        # before the interrupt comes, even in release mode on fast machines.
+        last_duration = 0.0
+        workload_size = 100_000
+
+        while last_duration < 1.0:
+            print("workload size:", workload_size)
```
Comment (cyb70289) [inline python/pyarrow/tests/test_csv.py:1329]: Looks pytest hides the output by default. Curious to see the workload size big enough for M1.

Comment (pitrou) [reply python/pyarrow/tests/test_csv.py:1329]: You can pass the `-s` option to pytest. Alternatively, if the test fails, the output is shown.

**Review by kszucs — APPROVED** [review]
Thanks Antoine!

**Pull request closed without merging**